src/main/java/seedu/address/model/addresses/PublicAddressesComposition.java

```diff
@@ -0,0 +1,257 @@
+
+    /**
+     * Constructs an empty PublicAddressesComposition.
+     */
+    public PublicAddressesComposition() {
+        this.publicAddresses = new HashMap<>();
+    }
+
+    /**
+     * Constructs a PublicAddressesComposition with the given map of public addresses.
+     *
+     * @param publicAddresses A map of networks to sets of public addresses.
+     */
+    public PublicAddressesComposition(Map<Network, Set<PublicAddress>> publicAddresses) {
+        this.publicAddresses = publicAddresses.entrySet().stream()
+                .collect(HashMap::new, (m, e) -> m.put(e.getKey(), new HashSet<>(e.getValue())), HashMap::putAll);
+    }
+
+    /**
+     * Adds a public address to the specified network. If the network does not exist, it is created.
+     *
+     * @param network       The network to which the public address belongs.
+     * @param publicAddress The public address to be added.
+     */
+    public void addPublicAddress(Network network, PublicAddress publicAddress) {
+        assert network != null;
+        assert publicAddress != null;
+        publicAddresses.computeIfAbsent(network, k -> new HashSet<>());
+        publicAddresses.get(network).add(publicAddress);
+    }
+
+    /**
+     * Sets the public addresses for the specified network.
+     *
+     * @param network         The network to set the public addresses for.
+     * @param publicAddresses The set of public addresses to be set.
+     */
+    public void setPublicAddressForNetwork(Network network, Set<PublicAddress> publicAddresses) {
+        assert network != null;
+        assert publicAddresses != null;
+        this.publicAddresses.put(network, new HashSet<>(publicAddresses));
+    }
+
+    /**
+     * Returns the map of public addresses.
+     *
+     * @return A map of networks to sets of public addresses.
+     */
+    public Map<Network, Set<PublicAddress>> getPublicAddresses() {
+        return publicAddresses;
+    }
+
+    /**
+     * Returns the set of public addresses for the specified network.
+     *
+     * @param network The network to get the public addresses for.
+     * @return An unmodifiable set of public addresses for the specified network.
+     */
+    public Set<PublicAddress> getByNetwork(Network network) {
+        assert network != null;
+        return Collections.unmodifiableSet(publicAddresses.getOrDefault(network, new HashSet<>()));
+    }
+
+    /**
+     * Adds a set of public addresses to the specified network.
+     *
+     * @param network   The network to add the public addresses to.
+     * @param addresses The set of public addresses to be added.
+     */
+    public void addPublicAddressesToNetwork(Network network, Set<PublicAddress> addresses) {
+        assert network != null;
+        assert addresses != null;
+        if (addresses.isEmpty()) {
+            this.publicAddresses.remove(network);
+            return;
+        }
+        this.publicAddresses.put(network, new HashSet<>(addresses));
+    }
```

Comment:
I think operations related to delete should be separate methods. The behavior does not match what the method name. Also duplicate with setPublicAddressForNetwork?

Reply:
addPublicAddressesToNetwork should adds the set of addresses without overriding the existing ones,
setPublicAddressesForNetwork replaces all the publicAddresses with the given set overriding the existing one